### PR TITLE
create curl task to request response from internet

### PR DESCRIPTION
**Pull request by yuege01**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by yuege01** [comment]
@wlynch @dibyom

**Comment by yuege01** [comment]
> Thanks for contributing this, any chance we can have the README/example and test included with this submission ?
> 
> thanks,

Sure

File: curl/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    - name: url
+      value: "www.google.com"
+    - name: options
+      value:
+       - "-v"
+  taskRef:
+    name: curl
+```
+
+Run it with `kubectl create -f curl-header-run.yaml`
```
Comment (dibyom) [inline curl/README.md:44]: same as https://github.com/tektoncd/catalog/pull/357#discussion_r437691470

either use `generateName` or `kubectl apply`

Comment (dibyom) [reply curl/README.md:44]: Hmm, also `curl -v` will print out both headers as well as the body. For only headers, we can use `curl -i`

File: curl/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/curl/curl.yaml
+```
+
+## Parameters
+
+* **urlimages/curl**: `curl` CLI container image to run this task.
```
Comment (dibyom) [inline curl/README.md:14]: Doesn't look like this is actually a param. Suggest we remove it!

File: curl/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/curl/curl.yaml
+```
+
+## Parameters
+
+* **options**: The arguments to pass to `curl` CLI.  _default_: `[""]`
+* **url**: The url we want to download file from.  _default_: `""`
```
Comment (chmouel) [inline curl/README.md:15]: I don't this it is `default: ""` but instead being _required_

**Comment by chmouel** [comment]
/lgtm


nice work on this, thanks!

File: curl/curl.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+   - name: url
+     description: URL to curl'ed
+     type: string
+   - name: options
+     description: options of url
+     type: array
+     default: []
+  steps:
+    - name: curl
+      image: curlimages/curl
```
Comment (wlynch) [inline curl/curl.yaml:16]: We should make this a param that defaults to `curlimages/curl`. This way users can pin to versions if needed.

Comment (yuege01) [reply curl/curl.yaml]: Thanks for the advice, I've changed my curl.yaml file and updated readme accordingly.

File: curl/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/curl/curl.yaml
+```
+
+## Parameters
+
+* **options**: The arguments to pass to `curl` CLI.  _default_: `[""]`
+* **url**: The url we want to download file from, required field.
+* **curl-image**: Option of curl image.
```
Comment (dibyom) [inline curl/README.md:16]:
```suggestion
* **curl-image**: The curl docker image to be used. Defaults to `curlimages/curl`
```

**Comment by yuege01** [comment]
> Could you squash up the commits? Otherwise, good to go!

Sure

**Comment by chmouel** [comment]
/lgtm

thank you,

**Pull request merged**